src/par_iter/mod.rs

```diff
@@ -468,35 +469,70 @@ pub trait ParallelIterator: Sized {
+    /// the results are not deterministic.
     ///
-    /// Basically equivalent to `self.reduce(|| MAX, |a, b| cmp::min(a, b))`
-    /// except that the type of `MAX` and the `min` operation may vary
-    /// depending on the type of value being produced.
-    fn min(self) -> Self::Item
-        where MinOp: ReduceOp<Self::Item>
+    /// Basically equivalent to `self.reduce_with(|a, b| cmp::min(a, b))`.
+    fn min(self) -> Option<Self::Item>
+        where Self::Item: Ord
     {
-        reduce(self, MIN)
+        self.reduce_with(cmp::min)
     }
 
-    /// Computes the maximum of all the items in the iterator.
+    /// Computes the item that yields the minimum value for the given
+    /// function. If the iterator is empty, `None` is returned;
+    /// otherwise, `Some(item)` is returned.
     ///
-    /// Note that the order in items will be reduced is not specified,
-    /// so if the `Ord` impl is not truly associative, then the
-    /// results are not deterministic.
+    /// Note that the order in which the items will be reduced is not
+    /// specified, so if the `Ord` impl is not truly associative, then
+    /// the results are not deterministic.
+    fn min_by_key<K, F>(self, f: F) -> Option<Self::Item>
+        where K: Ord,
+              F: Sync + Fn(&Self::Item) -> K,
+    {
+        self.reduce_with(|a, b| match f(&a).cmp(&f(&b)) {
+            Ordering::Less => a,
+            Ordering::Equal=> b,
+            Ordering::Greater => b,
+        })
+    }
+
+    /// Computes the maximum of all the items in the iterator. If the
+    /// iterator is empty, `None` is returned; otherwise, `Some(max)`
+    /// is returned.
     ///
-    /// Basically equivalent to `self.reduce(|| MIN, |a, b| cmp::max(a, b))`
-    /// except that the type of `MIN` and the `max` operation may vary
-    /// depending on the type of value being produced.
-    /// are not deterministic.
-    fn max(self) -> Self::Item
-        where MaxOp: ReduceOp<Self::Item>
+    /// Note that the order in which the items will be reduced is not
+    /// specified, so if the `Ord` impl is not truly associative, then
+    /// the results are not deterministic.
+    ///
+    /// Basically equivalent to `self.reduce_with(|a, b| cmp::max(a, b))`.
+    fn max(self) -> Option<Self::Item>
+        where Self::Item: Ord
+    {
+        self.reduce_with(cmp::max)
+    }
+
+    /// Computes the item that yields the maximum value for the given
+    /// function. If the iterator is empty, `None` is returned;
+    /// otherwise, `Some(item)` is returned.
+    ///
+    /// Note that the order in which the items will be reduced is not
+    /// specified, so if the `Ord` impl is not truly associative, then
+    /// the results are not deterministic.
+    fn max_by_key<K, F>(self, f: F) -> Option<Self::Item>
+        where K: Ord,
+              F: Sync + Fn(&Self::Item) -> K,
     {
-        reduce(self, MAX)
+        self.reduce_with(|a, b| match f(&a).cmp(&f(&b)) {
```

Comment:
I think it would be better if this only called `f` once per item.  Something like:
```rust
self.map(|x| (f(&x), x)).reduce_with(...).map(|(_, x)| x)
```
That will require `K: Send` though.